gym/core.py

```diff
@@ -242,18 +251,20 @@ class Wrapper(Env[ObsType, ActType]):
         Don't forget to call ``super().__init__(env)`` if the subclass overrides :meth:`__init__`.
     """
 
-    def __init__(self, env: Env):
+    def __init__(self, env: Env, new_step_api: bool = False):
```

Comment:
Personally, Im really unsure of this approach to adding new_step_api in wrappers. Even if we go with this approach, I don't like this code style as every wrappers needs to add this parameter. 
I can't remember our proposed method but I think that the whole API should support the new API with the compatibility wrapper to used to convert between the two.

Reply:
They don't need to override the `new_step_api` attribute if they're using the old style, but they do need if it will be in the new style (cuz we're doing opt-in). I understand there is some ugliness, but I don't know how to retain 100% backward compatibility while keeping it clean